Feat/bac 55 adding upload function data storage

devvspaces opened the pull request:
- Created file storage function to upload files to zc_core
- Handled all error edge cases
- Wrote tests for all cases

Linear: https://linear.app/team-socket/issue/BAC-55/implementing-the-data-storage-upload-function-to-upload-one-or-more

![pr](https://user-images.githubusercontent.com/77179231/204327218-ab0fb7c9-37f5-44d9-b4e4-0f3a5af7254e.png)

Bernadettechukwuedo committed:
feat(BAC-53):Endpoint for viewing the details of a particular file

Bernadettechukwuedo committed:
feat(BAC-53):Endpoint for viewing the details of a particular file

Commit:
feat(BAC-55): Added upload function in data storage class

- Upload function will be used by send file endpoint when created
- Upload function works with one or more files
- Handles error cases gracefully
- Wrote test to test all cases

Commit:
Merge branch 'dev' of https://github.com/zurichat/zc_messaging into feat/BAC-55-adding-upload-function-data-storage

FlorenceEgwu commented on backend/requirements.txt:
```diff
@@ -18,7 +18,7 @@ fastapi-pagination==0.9.1
 h11==0.12.0
 httpcore==0.15.0
-httptools==0.2.0
+httptools==0.5.0
```
take out the versioning on this  completely. I think this still gives issue for some people.
i.e we should have
```suggestion
httptools
```

FlorenceEgwu commented on backend/tests/utils/test_db.py:
```diff
@@ -14,3 +16,89 @@ async def test_connection_timeout():
+                    "status": 400,
+                    "data": "test"
+                }
+            }
```
great job on adding these test cases. Really happy with it.

FlorenceEgwu commented on backend/utils/db.py:
```diff
@@ -373,3 +377,59 @@ async def get_member(
+        headers = {
+            "Authorization": token,
+        }
+
```
I thought headers like this were handled somewhere central in this class.
@zxenonx , can you remember?

zxenonx replied:
@AI-fae, I remember we did this in zc_plugin_dm but not here.
Besides, we need to protect these endpoints now.

FlorenceEgwu commented on backend/utils/db.py:
```diff
@@ -373,3 +377,59 @@ async def get_member(
                 return member if member["_id"] == member_id else {}
+
+    async def upload(
+        self, files: List[Any], token: str
```
`files_upload` would be a more descriptive name for this method

devvspaces replied:
Okay thanks, I will make the required changes

FlorenceEgwu commented:
Great job on this.
I have added a few comments

Commit:
fixed: Resolved changes requested

- Updated requirements.txt by removing versioning from `httptools`
- Made function name more descriptive

zxenonx commented on backend/utils/message_utils.py:
```diff
@@ -1,4 +1,4 @@
-from typing import Any, Optional, List, Dict
+from typing import Any, Dict, List, Optional
```
We no longer need to import List and Dict from Python 3.9.
https://peps.python.org/pep-0585/
list and dict can be used. They are built in

zxenonx replied:
@devvspaces

zxenonx reviewed (approved):
Good job

Commit:
Merge branch 'dev' of https://github.com/zurichat/zc_messaging into feat/BAC-55-adding-upload-function-data-storage

Commit:
fix: BAC-55

- renamed List and Dict to use the list and dict as per python version upgrade allows it

Commit:
fix(BAC-55): Figured and tested the right way to call this api

- Found out that @zxenonx has a class for files called file storage.
- Realized that the file_upload function should be there
- I refactored the code, added the function to the file storage class.
- The function was tested with the initialization.
- The file upload is still working on the zc_core staging. Tested that with teammates

Commit:
Merge branch 'dev' of https://github.com/zurichat/zc_messaging into feat/BAC-55-adding-upload-function-data-storage

Commit:
fix: Removed file_upload functin from db.py

Commit:
fix: removed the upload_api property from db.py

mikengr reviewed (approved):
LGTM
